python/paddle/fluid/contrib/mixed_precision/fp16_utils.py

```diff
@@ -80,8 +80,9 @@ def create_master_params_grads(params_grads, main_prog, startup_prog,
         A list of master parameters and gradients. 
     """
     master_params_grads = []
-    with main_prog._backward_role_guard():
-        for p, g in params_grads:
+    for p, g in params_grads:
+        # create master parameters
+        with main_prog._optimized_guard([p, g]):
```

Comment:
what's the difference while training with pe? As the op orders in the front-end have no effect on the dependency.

Reply:
It doesn't change the dependency, but delay op's running to avoid waiting.